contrib/golang/common/go/api/filter.go

```diff
@@ -205,3 +205,16 @@ type FilterState interface {
 	SetString(key, value string, stateType StateType, lifeSpan LifeSpan, streamSharing StreamSharing)
 	GetString(key string) string
 }
+
+type ConfigCallbacks interface {
+	DefineMetric(metricType uint32, name string) Metric
```

Comment:
we should define a new type, i.e. MetricType, instead of  using the raw `uint32`. see  `StatusType` as an example.